### DISCUSSION TITLE
StructuredGrid scalars

### DISCUSSION THREAD

**Discussion by Unikeako** [pyvista/pyvista#4487] [Q&A] [answered]
Hi all,

I was trying to build a StructuredGrid from the Numpy array. The shape is as expected. I wanted to color it using the z column. Then I got a wired result as shown in p1.png. The color range seems right, but the mapping relation is incorrect. Then I used the example data from this link: https://docs.pyvista.org/version/stable/examples/00-load/create-structured-surface.html#sphx-glr-examples-00-load-create-structured-surface-py, I found that I could also get similar results as shown in p2.png when I changed the range of x and y, such as x = np.arange(0, 20, 1); y = np.arange(0, 30, 1). 
![p1](https://github.com/pyvista/pyvista/assets/131874962/cca551b1-6808-4847-a3b4-0815362e9be4)
![p2](https://github.com/pyvista/pyvista/assets/131874962/cdefc213-587d-44f9-b80d-f678d1dc177c)

I will be appreciated if anyone could help. 
Besides, can I make a VTK file including extra data except for the geometry data using pyvista?

Best regards
Yu

**Comment by banesullivan** [comment, marked as answer]
Ah yep this is an `indexing='ij'` vs `indexing='xy'` issue which @natsuwater recently improved in many other examples in https://github.com/pyvista/pyvista/pull/4365. 

When calling `np.meshgrid()` be sure to pas `indexing='ij'`

**Reply by banesullivan** [reply]
We should update this example to pass that option

**Reply by Unikeako** [reply]
Problem solved, thank you very much!
